### PR TITLE
Add poison wisp

### LINKED ISSUE #48: [FEATURE] poison wisp
# Describe the feature

poison wisp 

# Describe The needs

leaves a  poison trail

### PR DESCRIPTION
**Pull request by EnzoDrapron**
# Description

Add poison wisp who leaves a poison trail on the ground

Close #48 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Minor fixes

Remove Random color for AttackWisp
Pass _Color in wisp file to a Serialized Field

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Assets/Prefabs/Wisps/PoisonWisp.prefab [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+  m_IsActive: 1
+--- !u!4 &460567562440483175
+Transform:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 5846959578341104856}
+  m_LocalRotation: {x: 0, y: 0, z: 0, w: 1}
+  m_LocalPosition: {x: 1.128314, y: 0.452361, z: 0}
```
Comment (AndreasLrx) [inline Assets/Prefabs/Wisps/PoisonWisp.prefab:32]: Set local position to 0

File: Assets/Scripts/AttackWisp.cs [hunk 2]
```diff
@@ -7,20 +7,10 @@ public class AttackWisp : Wisp
...
-            Random.Range(0.4f, 1f)
-        );
-        ResetColor();
-    }
-
     protected override IEnumerator OnActivate()
     {
         SetTarget((Vector2)playerObject.transform.position + Player().AimedDirection() * range);
-        while (MoveTowardsTarget() && !Attack(wispDamage))
+        while (MoveTowardsTarget() && !Attack(wispDamage    ))
```
Comment (AndreasLrx) [inline Assets/Scripts/AttackWisp.cs:13]: Remove extra spaces

File: Assets/Scripts/PoisonWisp.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    void Awake() {
+        polygonCollider2D = GetValidCollider();
+        trailRenderer = this.GetComponent<TrailRenderer>();
+        trailRenderer.material.color = Color.green;
+
+        base.Awake();
+    }
+
+    void Update() {
+        SetColliderPointFromTrail(trailRenderer,polygonCollider2D);
```
Comment (AndreasLrx) [inline Assets/Scripts/PoisonWisp.cs:30]: missing space after coma

File: Assets/Scripts/PoisonWisp.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            validCollider.enabled = true;
+            unusedColliders.RemoveAt(0);
+        }else{
+            validCollider = new GameObject("TrailCollider",typeof(EdgeCollider2D)).GetComponentInParent<EdgeCollider2D>();
+            validCollider.GetComponent<EdgeCollider2D>().isTrigger = true;
+        }
+        return validCollider;
+    }
+
+    void SetColliderPointFromTrail(TrailRenderer trail,EdgeCollider2D collider){
```
Comment (AndreasLrx) [inline Assets/Scripts/PoisonWisp.cs:51]: missing space after coma

File: Assets/Scripts/PoisonWisp.cs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        List<Vector2> points = new List<Vector2>();
+        if(trail.positionCount == 0){
```
Comment (AndreasLrx) [inline Assets/Scripts/PoisonWisp.cs:53]: remove unused curly brackets

File: Assets/Scripts/PoisonWisp.cs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+    protected override IEnumerator OnAttach()
+    {
+        trailRenderer.startWidth = attachTrailWidth;
+        trailRenderer.time = attachTrailTime;
+        yield break;
+    }
+
+    private void OnDestroy() {
+        if(polygonCollider2D!=null){
```
Comment (AndreasLrx) [inline Assets/Scripts/PoisonWisp.cs:83]: Add space around operators

**Review by AndreasLrx — CHANGES_REQUESTED** [review]

**Review by AndreasLrx — APPROVED** [review]

**Review by cley44 — APPROVED** [review]

**Review by AndreasLrx — APPROVED** [review]

**Pull request merged**